server/modelpath.go

```diff
@@ -23,33 +24,37 @@ const (
 	DefaultProtocolScheme = "https"
 )
 
+var protocolPattern = regexp.MustCompile(`^.*://`)
+
 func ParseModelPath(name string) ModelPath {
+	name = protocolPattern.ReplaceAllString(name, "")
+
 	slashParts := strings.Split(name, "/")
-	var registry, namespace, repository, tag string
+
+	registry := DefaultRegistry
+	namespace := DefaultNamespace
+	repository := ""
+	tag := DefaultTag
 
 	switch len(slashParts) {
 	case 3:
 		registry = slashParts[0]
 		namespace = slashParts[1]
-		repository = strings.Split(slashParts[2], ":")[0]
+		repository = slashParts[2]
```

Comment:
Hello! Thanks for opening a PR 😊 

This is close, but I think you'll want to apply this to `registry` vs `repository`. I added an example here: https://github.com/jmorganca/ollama/issues/371